sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -175,6 +175,64 @@ object GpuCast extends Arm {
       cv.stringReplaceWithBackrefs(rule.search, rule.replace)
     })
   }
+
+  def sanitizeStringToFloat(input: ColumnVector): ColumnVector = {
+
+    // This regex gets applied after the transformation to normalize use of Inf and is
+    // just strict enough to filter out known edge cases that would result in incorrect
+    // values. We further filter out invalid values using the cuDF is_float method.
+    val VALID_FLOAT_REGEX =
+      "^" +                         // start of line
+      "[+\\-]?" +                   // optional + or - at start of string
+      "(" +
+        "(" +
+          "(" +
+            "([0-9]+)|" +           // digits, OR
+            "([0-9]*\\.[0-9]+)|" +  // decimal with optional leading and mandatory trailing, OR
+            "([0-9]+\\.[0-9]*)" +   // decimal with mandatory leading and optional trailing
+          ")" +
+          "([eE][+\\-]?[0-9]+)?" +  // exponent
+          "([fFdD])?" +             // floating-point designator
+        ")" +
+        "|(Inf)" +                  // Infinity
+        "|([nN][aA][nN])" +         // NaN
+      ")" +
+      "$"                           // end of line
+
+    withResource(input.lstrip()) { stripped =>
+      withResource(GpuScalar.from(null, DataTypes.StringType)) { nullString =>
+        // filter out strings containing breaking whitespace
+        val withoutWhitespace = withResource(stripped.containsRe("(\r|\n)")) {
+            hasVerticalWhitespace =>
+          hasVerticalWhitespace.ifElse(nullString, stripped)
+        }
+        // replace all possible versions of infinity with Inf
+        withResource(withoutWhitespace) { _ =>
+          withResource(withoutWhitespace.stringReplaceWithBackrefs(
+            "([iI][nN][fF])([iI][nN][iI][tT][yY])?", "Inf")) { inf =>
+            // replace +Inf with Inf
+            withResource(inf.stringReplaceWithBackrefs("([+])(Inf)", "Inf")) {
+                infWithoutPlus =>
+              // filter out any strings that are not valid floating point numbers
+              val floatOrNull = withResource(infWithoutPlus.matchesRe(VALID_FLOAT_REGEX)) {
+                  looksLikeFloat =>
+                looksLikeFloat.ifElse(infWithoutPlus, nullString)
+              }
+              // strip floating-point designator 'f' or 'd' but don't strip the 'f' from 'Inf'
+              withResource(floatOrNull) { _ =>
+                withResource(floatOrNull.matchesRe("^[+\\-]?Inf$")) { isInf =>
```

Comment:
There isn't a regex that can strip the trailing designator without needing to check for `Inf` separately first?  I think we just want a string regex replace of `(^n)[fFdD]` with `\1`.